src/yki/boundary/exam_date_db.clj

```diff
@@ -17,33 +17,48 @@
 
 (defprotocol ExamDate
   (create-exam-date! [db exam-date])
-  (create-exam-date-languages! [db exam-date-id languages])
+  (update-exam-date! [db id exam-date-update new-languages removable-languages])
   (get-exam-dates [db])
   (get-organizer-exam-dates [db from])
   (get-exam-date-by-id [db id])
   (get-exam-date-session-count [db id])
   (get-exam-dates-by-date [db exam-date])
-  (get-exam-date-language [db id language])
   (get-exam-date-languages [db id])
-  (delete-exam-date! [db id])
-  (delete-exam-date-languages! [db id languages])
-  (create-and-delete-exam-date-languages! [db id new-languages removed-languages])
-  (update-post-admission-details! [db id post-admission])
-  (toggle-post-admission! [db id enabled]))
+  (delete-exam-date! [db id]))
 
 (extend-protocol ExamDate
   Boundary
   (create-exam-date!
     [{:keys [spec]} exam-date]
     (jdbc/with-db-transaction [tx spec]
       (rollback-on-exception
-       tx
-       #(let [converted-dates (convert-dates exam-date)
-              result (q/insert-exam-date<! tx converted-dates)
-              exam-date-id (:id result)]
+        tx
+        #(let [result (q/insert-exam-date<! tx {:exam_date              (string->date (:exam_date exam-date))
+                                               :registration_start_date (string->date (:registration_start_date exam-date))
+                                               :registration_end_date   (string->date (:registration_end_date exam-date))})
+               id (:id result)]
           (doseq [lang (:languages exam-date)]
-            (q/insert-exam-date-language! tx (assoc lang :exam_date_id exam-date-id)))
-          exam-date-id))))
+            (q/insert-exam-date-language! tx (assoc lang :exam_date_id id)))
+          id))))
+  (update-exam-date!
+    [{:keys [spec]} id exam-date-update new-languages removable-languages]
+    (jdbc/with-db-transaction [tx spec]
+      (rollback-on-exception
+        tx
+        #(let [_ (q/update-exam-date! tx {:id                        id
+                                          :exam_date                 (string->date (:exam_date exam-date-update))
+                                          :registration_start_date   (string->date (:registration_start_date exam-date-update))
+                                          :registration_end_date     (string->date (:registration_end_date exam-date-update))
+                                          :post_admission_start_date (string->date (:post_admission_start_date exam-date-update))
+                                          :post_admission_end_date   (string->date (:post_admission_end_date exam-date-update))
```

Comment:
Ei sinänsä mitään väärää tässä, mutta mapin voisi rakentaa myös esim. seuraavasti: 
```
(-> exam-date-update
    (select-keys
     [:exam_date
      :registration_start_date
      :registration_end_date
      :post_admission_start_date
      :post_admission_end_date])
    (update-vals string->date)
    (assoc :id id :post_admission_enabled (:post_admission_enabled exam-date-update)))
```

Ei sinänsä tarvii muuttaa, mutta ajattelinpa mainita tämänkin vaihtoehdon.

Reply:
Ehkä jätän alustavasti näin kun create tehty myös nykyisellä mallilla ja siinä vain kolme muuttujaa, jotka mäpättäisiin tuolla string->date funktiolla ja siihen ehkä tällainen malli vähän tarpeeton